File: debug-adapter/sampleWorkspace/binary_search.cpp
// write a program for binary search
#include <iostream>
using namespace std;

int binarySearch(int arr[], int n, int key)
{
  int s = 0;
  int e = n;
  while (s <= e)
  {
    int mid = (s + e) / 2;
    if (arr[mid] == key)
      return mid;
    else if (arr[mid] > key)
      e = mid - 1;
    else
      s = mid + 1;
  }
  return -1;
}

int main()
{
  int arr[] = { 2, 3, 4, 10, 40 };
	int x = 10;
	int n = sizeof(arr) / sizeof(arr[0]);
	int result = binarySearch(arr, n - 1, x);
	(result == -1)
		? cout << "Element is not present in array"
		: cout << "Element is present at index " << result;
	return 0;
}

File: debug-adapter/sampleWorkspace/binary_search.py
# make an array of 100 random numbers

import random
arr = [random.randint(1, 100) for i in range(100)]
arr.sort()

# define a function to search for a number in the array
def binary_search(arr, x):
    l = 0
    r = len(arr) - 1
    while l <= r:
        mid = (l + r) // 2
        if arr[mid] == x:
            return mid
        elif arr[mid] < x:
            l = mid + 1
        else:
            r = mid - 1
    return -1

# search for a number in the array
x = 50
result = binary_search(arr, x)
if result != -1:
    print(f"Element is present at index {result}")
else:
    print("Element is not present in array")